### ISSUE TITLE
Extend abilities of #[JsonProperty]

### ISSUE THREAD

**Issue by JuraSciix** [JuraSciix/Objeckson#2]
By specifying #[JsonProperty] over a entire class or enum, the attribute distributes to all members of the class.

Example:
```php

#[JsonProperty]
enum Status: int {
    case PASSED = 1;
    case NOT_PASSED: 0;
}

#[JsonProperty]
class Rate {
    public string $subject;
    public Status $status;
}

#[JsonProperty]
class Student {
    /**
     * @var Rate[]
     */
    public array $rates;
}

$data = [
    'rates' => [
        [
            'subject' => 'Math',
            'status' => 1
        ]
    ]
];
$objeckson = new Objeckson();
$student = $objeckson->fromJson($data, Student::class);
var_dump($student);
// Works fine.
```

**Comment by JuraSciix** [comment]
Implemented in 90aa48c1c7696f51e8ce52f95e2708d6832f507a.

**Comment by JuraSciix** [comment]
Implemented in 90aa48c1c7696f51e8ce52f95e2708d6832f507a.
